### ISSUE TITLE
Segmentation fault in x86_64 Linux container running on Apple M1 hardware

### ISSUE THREAD

**Issue by dan-cohn-sabre** [hadolint/hadolint#823]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior

Hadolint runs properly.

### Actual behavior

```bash
$ hadolint Dockerfile
qemu: uncaught target signal 11 (Segmentation fault) - core dumped
Segmentation fault
```

### Steps to reproduce the behavior

1. Run Linux x86_64 Docker container on Mac with M1 hardware.

2. Install hadolint by downloading hadolint-Linux-x86_64 from GitHub and making the file executable. We tried it with hadolint v2.8.0 and v2.10.0.

3. Run hadolint on a Dockerfile.

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter 2.10.0
```

**Dockerfile** (if relevant)

```Dockerfile
(paste content of your Dockerfile)
```

**Additional environment details (OS, stack version, etc.)**

Host processor model: Apple M1 Pro
Host OS: MacOS Monterey 12.3.1
Container OS: Centos 8 Stream (x86_64 GNU/Linux)

**Comment by lorenzo** [comment]
better run the arm64 image?

**Comment by jan-xyz** [comment]
Where do I find the Darwin arm64 image? I can only see the Linux image on the release page.

**Comment by lorenzo** [comment]
@jan-xyz the latest version's image works for both amd64 and amd64, just use it as normal like this:

```
docker run --rm ghcr.io/hadolint/hadolint:v2.10.0 hadolint --version
```

**Comment by jan-xyz** [comment]
A I missed that this is about the Docker image. I thought we were talking about the binary release. 🤦

**Comment by dan-cohn-sabre** [comment]
@lorenzo I don't think you understand my use case. I need to run hadolint inside an x86 (AMD64) container. It's part of a set of commit hooks that we run within a container-based development environment.

The problem is that the x86_64 version of hadolint fails to run when the Docker include an emulation layer to enable an AMD64-based image to run on ARM64 hardware.

**Comment by dan-cohn-sabre** [comment]
> A I missed that this is about the Docker image. I thought we were talking about the binary release. 🤦

Just to be clear, I am talking about the binary release running within a custom Docker image (not the hadolint image).

**Comment by lorenzo** [comment]
@dan-cohn-sabre ok, I understand now.  I also have an M1 and I know how many x86 containers end up having weird qemu errors.

My only hunch is that the problem could be caused by the compression program `upx`, which we use for releases. Other than that, I would not even know where to start investigating that fault.

**Comment by dan-cohn-sabre** [comment]
@lorenzo Can you reopen the issue? My team is looking into this also and may be able to help find the root cause.

**Comment by viktorsboroviks** [comment]
Still encounter same issue on M2.
Only solution that worked was to build hadolint in docker myself and run from there.
